tests/pytests/functional/loader/test_loader.py

```diff
@@ -1,14 +1,23 @@
 import json
+import os
 
 import pytest
 
 from salt.utils.versions import Version
 from tests.support.helpers import SaltVirtualEnv
 from tests.support.pytest.helpers import FakeSaltExtension
+from tests.support.runtests import RUNTIME_VARS
+
+MISSING_SETUP_PY_FILE = not os.path.exists(
+    os.path.join(RUNTIME_VARS.CODE_DIR, "setup.py")
+)
 
 pytestmark = [
     # These are slow because they create a virtualenv and install salt in it
     pytest.mark.slow_test,
+    pytest.mark.skipif(
+        MISSING_SETUP_PY_FILE, reason="This test only work if setup.py is available"
+    ),
```

Comment:
Can we instead supply a `setup.py`? The loader is a core component, it would be good to test it

Reply:
It seems it is not enough to provide an example `setup.py` file, neither the `setup.py` from Salt.

These tests are expecting that `salt` is not yet built when running the test (it is expecting the `setup.py` and other build artifacts to be there), as the tests triggers the "salt" building & installation inside a test "virtualenv".

Also the fact that we have splitted our Salt package installation path from our salt testsuite path is also making the tests to mess up.

OTOH these tests are actually running in GH actions, as Salt is not yet built. The tests would be skipped only when running them in a packaged Salt setup.

Reply:
> OTOH these tests are actually running fine in GH actions, as Salt is not yet built. The tests would be skipped only when running them in a packaged Salt setup.

That by itself is not very convincing, the packaged Salt setup is the setup that's more important to test.

> These tests are expecting that salt is not yet built when running the test (it is expecting the setup.py and other build artifacts to be there), as the tests triggers the "salt" building & installation inside a test "virtualenv".

We might need to revisit this for 3008. From what I can tell, the test covers loading salt extensions and that will be important to cover well.

I'm okay with skipping it for now though. Would be a candidate for the skiplist.

Reply:
Agree, we should ensure these salt extensions tests are running once we go for 3008.

For now, I would rather vote to keep this patch in the codebase instead of adding it to the current `skiplist`, as skiplist will ignore this test in all cases, not only when "setup.py" is missing.